apps/ai-content-generator/src/components/app/sidebar/feature-button/FeatureButton.tsx

```diff
@@ -1,31 +1,54 @@
 import { useSDK } from '@contentful/react-apps-toolkit';
 import { SidebarAppSDK } from '@contentful/app-sdk';
-import { Button, Tooltip } from '@contentful/f36-components';
+import { Button, Spinner, Tooltip } from '@contentful/f36-components';
 import featureConfig, { AIFeature } from '@configs/features/featureConfig';
 import { styles } from './FeatureButton.styles';
 import { makeDialogConfig } from '@configs/dialog/dialogConfig';
+import { useState } from 'react';
 
 interface Props {
   feature: AIFeature;
+  isSaving: boolean;
+  setIsSaving: (isSaving: boolean) => void;
```

Comment:
This is a general anti-pattern for statement management functions. It's better to use the handleFunctionName and onFunctionName convention` and separating your state setters and your functions and use the functions to be the objects you pass as props (aka callback functions).

Reply:
Thank you, I was moving too fast on this. I appreciate the catch!